### PR TITLE
add whitebox recipe

### PR DESCRIPTION
**Pull request by giswqs**
add whitebox recipe

### CODE REVIEW

**Comment by giswqs** [comment]
AppVeyor build keeps failing. Not sure how to resolve this. Any help will be appreciated. 

```
Resource usage statistics from testing whitebox:
   Process count: 2
   CPU time: Sys=0:00:00.1, User=0:00:00.2
   Memory: 26.1M
   Disk usage: 467B
   Time elapsed: 0:00:06.0
TEST END: C:\bld\win-64\whitebox-0.5.0-py37_0.tar.bz2
Renaming work directory,  C:\bld\whitebox_1543038089058\work  to  C:\bld\whitebox_1543038089058\work_moved_whitebox-0.5.0-py>34_0_win-64_main_build_loop
Traceback (most recent call last):
  File "C:\Miniconda36-x64\lib\shutil.py", line 544, in move
    os.rename(src, real_dst)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\bld\\whitebox_1543038089058\\work' -> 'C:\\bld\\whitebox_1543038089058\\work_moved_whitebox-0.5.0-py>34_0_win-64_main_build_loop'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File ".ci_support\build_all.py", line 82, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 72, in build_all
    conda_build.api.build(list(d.keys()), config=config)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\api.py", line 207, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\build.py", line 2365, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda36-x64\lib\shutil.py", line 555, in move
    symlinks=True)
  File "C:\Miniconda36-x64\lib\shutil.py", line 315, in copytree
    os.makedirs(dst)
  File "C:\Miniconda36-x64\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\bld\\whitebox_1543038089058\\work_moved_whitebox-0.5.0-py>34_0_win-64_main_build_loop'
Command exited with code 1
```

**Comment by giswqs** [comment]
The whitebox package supports python>=3.5. When I specified `python >=3.5` under host and run of the requirements section, AppVeyor build failed. Therefore, I changed it to `python =3.6`

File: recipes/whitebox/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - click >=6.0
+    - pip
+    - python =3.6
+    - setuptools
+  run:
+    - click >=6.0
+    - python =3.6
```
[the remark this answers is not in the document]
Comment (giswqs) [reply recipes/whitebox/meta.yaml]: I changed it to `python >=3.6`, but appveyor failed to build. It built successfully when using `python =3.6`. Any suggestions? 

```
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File ".ci_support\build_all.py", line 134, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, old_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 124, in build_folders
    conda_build.api.build(list(d.keys()), config=get_config(arch, channel_urls))
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\build.py", line 2376, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda36-x64\lib\shutil.py", line 555, in move
    symlinks=True)
  File "C:\Miniconda36-x64\lib\shutil.py", line 315, in copytree
    os.makedirs(dst)
  File "C:\Miniconda36-x64\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\bld\\whitebox_1544031431233\\work_moved_whitebox-0.5.0-py>=36_0_win-64_main_build_loop'
Command exited with code 1
```

**Comment by beckermr** [comment]
@conda-forge/staged-recipes Is the windows failure a `conda-build` bug? 

Apparently the `>=` python version selector is not a valid set of characters for windows paths (https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names).

**Comment by giswqs** [comment]
Yes, I think it is a windows path issue. There must be some other conda Python packages that require `python >=3.x`. I was curious how they passed the appveyor windows build.

**Comment by isuruf** [comment]
See https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L32

**Comment by isuruf** [comment]
Changed your recipe to noarch, which should work fine with version constraints on python. If it's not a noarch recipe, version constraints have to be given using skip statements.

**Comment by giswqs** [comment]
@isuruf Great to know! Thanks a lot for your help. It is ready to be merged now.

**Pull request merged**